### ISSUE TITLE
--lr-scheduler flag does not work other than "polynomial_decay"

### ISSUE THREAD

**Issue by rnyak** [pytorch/fairseq#1587]
## 🐛 Bug
Hello,

I am finetuning Roberta-large model for text classification task with the  following command:
```
!CUDA_VISIBLE_DEVICES=0,1,2,3 python ./train.py /workspace/fairseq/data/ --restore-file $ROBERTA_PATH --max-positions 512 --max-sentences $MAX_SENTENCES --max-tokens 5200 --task sentence_prediction --reset-optimizer --reset-dataloader --reset-meters --required-batch-size-multiple 1  --init-token 0 --separator-token 2 --arch roberta_large --criterion sentence_prediction --classification-head-name $HEAD_NAME --num-classes $NUM_CLASSES --dropout 0.1 --attention-dropout 0.1 --weight-decay 0.1 --optimizer adam --adam-betas "(0.9, 0.98)" --adam-eps 1e-06 --clip-norm 0.0 --lr-scheduler polynomial_decay --lr $LR --total-num-update $TOTAL_NUM_UPDATES --warmup-updates $WARMUP_UPDATES --fp16 --fp16-init-scale 4 --threshold-loss-scale 1 --fp16-scale-window 128 --max-epoch 3 --best-checkpoint-metric accuracy --maximize-best-checkpoint-metric --truncate-sequence --find-unused-parameters --update-freq 4 --save-dir "/workspace/result/checkpoints/" 

```
That works without issue, but when I want to change the learning rate scheduler param as  `--lr-scheduler fixed`  or  `--lr-scheduler reduce_lr_on_plateau` it yields the following error:

```
train.py: error: unrecognized arguments: --total-num-update 330

```
How this can be fixed?

Thanks.

**Comment by myleott** [comment]
You need to remove the `--total-num-update` flag, which only works with the polynomial LR scheduler.
